4_de_analysis/nbconverted/1_de_control.py

```diff
@@ -0,0 +1,280 @@
+    print(str(save_file))
+    labels_df.to_csv(save_file, sep='\t')
+
+
+# In[16]:
+
+
+input_metadata_dir = str(cfg.de_input_dir)
+
+
+# In[17]:
+
+
+get_ipython().run_cell_magic('R', '-i RUN_DE_ANALYSIS -i base_dir -i input_str -i n_samples -i input_metadata_dir -i output_dir', "\nif (RUN_DE_ANALYSIS) {\n    source(paste0(base_dir, '/de_analysis.R'))\n\n    for (i in 0:(n_samples-1)) {\n        print(paste('Running: ', i))\n        input_metadata_str <- paste(\n            input_metadata_dir, '/lgg_idh1_random_s', i, '.tsv',\n            sep=''\n        )\n        get_DE_stats_DESeq(input_metadata_str,\n                           input_str,\n                           paste('LGG_IDH1_random_s', i, sep=''),\n                           output_dir)\n    }\n} else {\n    print('Skipping DE analysis, will use existing results files')\n}")
+
+
+# ### Compare IDH1 mutation DE results to randomly sampled results
+
+# In[18]:
+
+
+idh1_de_results_df = pd.read_csv(
+    cfg.de_output_dir / 'DE_stats_LGG_IDH1.txt',
+    sep='\t'
+)
+
+print(idh1_de_results_df.shape)
+idh1_de_results_df.head()
+
+
+# In[19]:
+
+
+random_de_results = []
+for i in range(n_samples):
+    
+    random_de_results.append(
+        pd.read_csv(
+            cfg.de_output_dir / 'DE_stats_LGG_IDH1_random_s{}.txt'.format(i),
+            sep='\t'
+        )
+    )
+    
+print(random_de_results[0].shape)
+random_de_results[0].head()
+
+
+# In[20]:
+
+
+# adjusted p-value threshold
+alpha = 0.05
+
+idh1_de_count = (
+    (idh1_de_results_df.padj < alpha).sum()
+)
+
+random_de_count = [
+    (random_de_results[ix].padj < alpha).sum() for ix in range(n_samples)
+]
+
+print('DE genes for IDH1 WT vs. mutant:', idh1_de_count)
+print('DE genes for random size-matched samples:', random_de_count)
+
+
+# In[21]:
+
+
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+sns.set({'figure.figsize': (8, 6)})
+
+sns.kdeplot(data=idh1_de_results_df.pvalue, label='true')
+for ix in range(n_samples):
+    if ix == 0:
+        sns.kdeplot(data=random_de_results[ix].pvalue,
+                    color='red', label='random')
+    else:
+        sns.kdeplot(data=random_de_results[ix].pvalue, color='red')
+plt.title('Uncorrected p-value density distributions')
+plt.xlabel('uncorrected p-value')
+plt.legend()
+
+
+# In[22]:
+
+
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+sns.set({'figure.figsize': (8, 6)})
+
+sns.kdeplot(data=idh1_de_results_df.padj, label='true')
+for ix in range(n_samples):
+    if ix == 0:
+        sns.kdeplot(data=random_de_results[ix].padj,
+                    color='red', label='random')
+    else:
+        sns.kdeplot(data=random_de_results[ix].padj, color='red')
+plt.title('FDR corrected p-value density distributions')
+plt.xlabel('Corrected p-value')
+plt.legend()
+
+
+# We can see that there are far more differentially expressed genes between IDH1 mutants and IDH1 wild-type samples than there are between random sample groups, which is what we expected.
```

Comment:
This description makes more sense to me. I was a bit confused by the wording at the header of this notebook - so you're basically comparing the number of DEGs identified using true labeling vs random sample labeling?

Reply:
What is this control for?

Reply:
> so you're basically comparing the number of DEGs identified using true labeling vs random sample labeling?

Yep, exactly

> What is this control for?

The idea for this experiment was basically to use the number of DE genes between two samples as a way to measure the similarity of (or difference between) the samples; more DE genes = less similarity.

So for the control, I took two cancer subtypes that I knew would be quite different (IDH1 mutant LGG has a much better prognosis than IDH1 wild-type LGG, and it's really easy to predict IDH1 mutation status from gene expression) and I counted the DE genes between them. To make sure that we see "a lot" of DE genes (which is what we expected since they're very different subtypes), I took a few random partitions of the same size from the same LGG data, without considering mutation status at all, and counted the DE genes between the partitions.

We saw > 10000 DE genes for the IDH1 mutant vs. WT comparison, and on the order of 100 DE genes for the random partitions (probably some of your generic/common DE genes), which is what we expected. This mostly just told me that the DE pipeline was implemented correctly, it wasn't a very "interesting" control other than that.